### PR TITLE
07-add-category_management-user_use_cases

### PR DESCRIPTION
**Pull request by andrey-kiyas**
Added:
- category management
- user use cases

### CODE REVIEW

File: src/main/java/com/onlinebookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
-@EnableMethodSecurity
+@EnableWebSecurity
+//@EnableMethodSecurity
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/config/SecurityConfig.java]: ??

File: src/main/java/com/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -29,22 +29,22 @@ public class BookController {
     private final BookService bookService;
 
     @Operation(summary = "Find all books", description = "Find all books")
-    @PreAuthorize("isAuthenticated()")
+    //@PreAuthorize("isAuthenticated()")
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/controller/BookController.java]: if you do not need it - delete it

File: src/main/java/com/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -32,6 +38,13 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/model/Book.java]: do not use EAGER
use @EntityGraph in repository

File: src/main/java/com/onlinebookstore/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @Column(name = "name", nullable = false, unique = true)
+    private String name;
+
+    private String description;
+
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(name = "books_categories",
+            joinColumns = @JoinColumn(name = "category_id"),
+            inverseJoinColumns = @JoinColumn(name = "book_id"))
+    private Set<Book> books;
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/model/Category.java]: we can not manage bidirectional connection this way - see this 
image

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/onlinebookstore/model/Category.java (hunk 4 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/model/Category.java]: I've had to verridd method "findAll" in "BookRepository" with "@EntityGraph" annotation

File: src/main/java/com/onlinebookstore/model/Book.java (hunk 3 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/model/Book.java]: Ok, done

File: src/main/java/com/onlinebookstore/controller/BookController.java (hunk 2 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/controller/BookController.java]: Ok, done

File: src/main/java/com/onlinebookstore/config/SecurityConfig.java (hunk 1 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/config/SecurityConfig.java]: Removed )

**Review by aaananas — APPROVED** [review]

**Pull request merged**